### DISCUSSION TITLE
DICOM Network support

### DISCUSSION THREAD

**Discussion by VitaliiLakusta** [suyashkumar/dicom#246] [Ideas]
Go currently misses a reliable dicom networking module. I am interested in developing and using a reliable module for DICOM networking (client/server C-STORE, C-MOVE etc.). Any willingness to discuss the idea further?

I am currently using https://github.com/grailbio/go-netdicom, but it's archived, and unfortunately has some bugs and isn't as complete and reliable as dcm4che or pynetdicom alternatives. Would be great to have a reliable dicom network support in Go.

**Comment by cajund** [comment]
Hi there,

I, too, would like to use this in the manner that you describe. Have you any simple examples that have given you success? I see that you have posted this some time ago, but if this is still an active project for you, I would be interested in collaborating.

Thanks.

**Comment by SemyonDunaev** [comment]
i'am interested too.

**Comment by cajund** [comment]
@VitaliiLakusta - Coming back around to this, as I have done considerable work in this area:

First. there are four repos that I have found to be branched from https://github.com/grailbio/go-netdicom with varying benefits. They are:

https://github.com/segmed/go-netdicom/
https://github.com/msz-kp/go-netdicom

These two have moved furthest from @grailbio's work, have have a number of improvements, specifically some work improving memory management and streaming from the network connection. it looks like @segmed did most of the improvements, with @msz-kp branching recently - these can be merged manually with little effort.

https://github.com/antibios/go-netdicom
https://github.com/mlibanori/go-netdicom

These two attempt to refactor the @grailbio repo to incorporate the improvement brought by @suyashkumar and this repo, however, they fork directly from grailbio, missing a lot of improvements from the above efforts. However, @antibios has tried to roll in this repo into the network, but are using cloned copies of the dicom library that aren't consistent with the published code. The work done by @mlibanori significantly refactors the original code, which I greatly appreciate, but it also misses the work done by the first two.

In my project, I have rolled in @msz-kp's work into my code base (not maintaining it as a separate module), as I have specific needs to the use case that are not represented in the code. I have started an effort to merge all four of these repos in my code base. I have also had to make some minor alterations to @suyashkumar's code to support some specifics, but all minor (to be posted in a separate issue)

I anticipate publishing the results as a separate repo in the near future, but if there are specific questions that you are anyone else has, I may be able to answer them.

Best.
